templates/client-daemonset.yaml

```diff
@@ -11,6 +11,12 @@ metadata:
     heritage: {{ .Release.Service }}
     release: {{ .Release.Name }}
 spec:
+  {{- if .Values.global.updateStrategy.enabled }}
+  updateStrategy:
```

Comment:
Lets just make this a YAML string because people might want to set `type: OnDelete` in which case they don't need the `rollingUpdate` key. It will be easier if we just template a yaml string.